Jenkinsfile

```
@@ -192,21 +193,43 @@ pipeline {
                   returnStdout: true
                 )
                 KUBECONFIG = "${env.WORKSPACE}/${e2e_environment}/modules/k8s/secrets/admin.conf"
+                TESTPLAN = "MQ-1"
               }
               steps {
-                // FIXME(arne-rusek): move hcloud's config to top-level dir in TF scripts
-                sh """
-                  mkdir -p "${e2e_environment}/modules/k8s/secrets"
-                """
-                copyArtifacts(
-                    projectName: "${k8s_job.getProjectName()}",
-                    selector: specific("${k8s_job.getNumber()}"),
-                    filter: "${e2e_environment}/modules/k8s/secrets/admin.conf",
-                    target: "",
-                    fingerprintArtifacts: true
-                )
-                sh 'kubectl get nodes -o wide'
-                sh "nix-shell --run './scripts/e2e-test.sh --device /dev/sdb --tag \"${env.GIT_COMMIT_SHORT}\" --registry \"${env.REGISTRY}\"'"
+                script {
+                  // FIXME(arne-rusek): move hcloud's config to top-level dir in TF scripts
+                  sh """
+                    mkdir -p "${e2e_environment}/modules/k8s/secrets"
+                  """
+                  copyArtifacts(
+                      projectName: "${k8s_job.getProjectName()}",
+                      selector: specific("${k8s_job.getNumber()}"),
+                      filter: "${e2e_environment}/modules/k8s/secrets/admin.conf",
+                      target: "",
+                      fingerprintArtifacts: true
+                  )
+                  sh 'kubectl get nodes -o wide'
+                  sh 'mkdir -p ./e2e_junit'
+                  def pd = sh(script: 'pwd', returnStdout:true).trim()
+                  def junitdir = "${pd}/e2e_junit"
+                  sh "nix-shell --run './scripts/e2e-test.sh --device /dev/sdb --tag \"${env.GIT_COMMIT_SHORT}\" --registry \"${env.REGISTRY}\" --reportsdir \"${junitdir}\"'"
```

```suggestion
                  sh """
                    mkdir -p e2e_junit
                    nix-shell --run './scripts/e2e-test.sh --device /dev/sdb --tag "${env.GIT_COMMIT_SHORT}" --registry "${env.REGISTRY}" --reportsdir e2e_junit'
                  """
```

This way you could also remove the additional `script {` nesting level as IMHO is not needed anymore.

If `e2e-test.sh` requires absolute path, than that's a bug in the script as I imagine one can run it manually and it should work with any user specified path. If that is indeed the case, please call `realpath` on `--reportsdir` value in the script itself. This looks like hot-fixing the `e2e-script.sh` in a wrapper.

---

Talking to Blaise about this, I think I'll avoid the use of a subdirectory altogether as it sounds preferable to keep the xml files in the default location, which is the repository root.